### DISCUSSION TITLE
Merge all event method into one simple method

### DISCUSSION THREAD

**Discussion by ChristopherGabba** [TheWidlarzGroup/react-native-video#4966] [Ideas]
### What problem are you trying to solve?

Tracking the player's status currently requires several separate events, each doing a specific thing:

- `onBuffer` : tracks whether or not the player is buffering
- `onPlaybackStateChanged` : tracks whether or not the player is playing
- `onEnd`: tracks whether or not the video is ended
- `onIdle`: currently missing from the documentation, but guess it's when the player idles

Would really simplify my code. Instead I have to have 4 different methods to handle this scenario. Would be great to have one single method that tracks the current status of the player.

### Proposed solution

Merge all of these event methods into a single method that tracks the current status of the player, like the `react-native-youtube-iframe` [onChangeState](https://lonelycpp.github.io/react-native-youtube-iframe/component-props#onchangestate).

#### Code sample

```typescript
const onChangeState = useCallback((state: VideoState) =>{
   switch(state):
      case "unstarted":
     // ... handle state
      case "buffering":
     // ... handle state
      case "paused":
     // ... handle state
},[])
```

#### Possible implementation

Not sure :(

### Additional context

Reference: `react-native-youtube-iframe` [onChangeState](https://lonelycpp.github.io/react-native-youtube-iframe/component-props#onchangestate)

**Comment by KrzysztofMoch** [comment]
Sound like good idea! I will add this to roadmap 👍

**Comment by zoriya** [comment]
I'd love if we could get that before the 7.0, deprecate old events and remove them in v7.

`onPlaybackStateChanged` is triggered when the player buffers BEFORE `onBuffer` is called. To prevent pausing (and updating controls) when the player buffers, the only solution is to use a setTimeout() and check if the player has buffered.

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
